### ISSUE TITLE
Text Contrast Errors for Foreground and Background Colors 

### ISSUE THREAD

**Issue by mtribone** [samvera/hyrax#1962]
### Descriptive summary

In Hyrax 2.0.0 rc1 and on the dashboard, there are several text contrast errors for the foreground and background colors.

### Rationale

Adequate contrast is necessary for all users, especially users with low vision or forms of color blindness.

### Expected behavior

Expected at least AA Compliance of all text colors.
https://webaim.org/standards/wcag/checklist#sc1.4.3

### Actual behavior

Text color in banner, breadcrumb link color and text color , previous and next link color for pagination, search placeholder text, transfer link text don't have sufficient contrast. 

![screen shot 2017-10-25 at 2 13 56 pm](https://user-images.githubusercontent.com/4163828/32015541-628f4d02-b98f-11e7-880a-8d9966178332.png)

### Steps to reproduce the behavior

1. Run an accessibility validator or color contrast validator
1. Review report on contrast errors

**Comment by mjgiarlo** [comment]
@mtribone Can you suggest colors for these changes?

**Comment by mtribone** [comment]
The purple banner at the top will pass AA if we darken it a bit (#6b6bb6). The same goes for the foreground color of the breadcrumb text, pagination text of prveious/next, and search for a user or placeholder text in general (#686868).  The linked home and select works to transfer seem to be false positives and are good to go.

**Comment by h-parekh** [comment]
1. Text color in banner:
  - Cannot reproduce in Vanilla Hyrax, this is specific to Nurax. 
  - `TODO`: change navbar to a darker shade in Nurax

2. breadcrumb link color (Home `<a>` tag)
  - This is meeting the contrast ratio for WCAG AA and AAA standards (See attached screenshot)
  - `TODO`: @mtribone  Is this a false alarm?

3. breadcrumb link color and text color
- `TODO`: Make the color of the text darker. Text color is being set by bootstrap CSS v3.3.7 in application.css
```css
# Current:
.breadcrumb>.active {
    color: #777777;
}
# Change to:
.breadcrumb>.active {
    color: #494949;
}
```
4. previous and next link color for pagination
  - `TODO`: Make the color of the text darker. Text color is being set by bootstrap CSS v3.3.7 in application.css
```css
# Current:
.pagination>.disabled>span, .pagination>.disabled>span:hover, .pagination>.disabled>span:focus, .pagination>.disabled>a, .pagination>.disabled>a:hover, .pagination>.disabled>a:focus {
    color: #777777;
    background-color: #fff;
    border-color: #ddd;
    cursor: not-allowed;
}
# Change to:
.pagination>.disabled>span, .pagination>.disabled>span:hover, .pagination>.disabled>span:focus, .pagination>.disabled>a, .pagination>.disabled>a:hover, .pagination>.disabled>a:focus {
    color: #595959;
    background-color: #fff;
    border-color: #ddd;
    cursor: not-allowed;
}
```
5. search placeholder text
  - `TODO`: Make the color of the text darker. Text color is being set by bootstrap CSS v3.3.7 in application.css
```css
# Current:
.select2-default {
    color: #999 !important;
}
# Change to:
.select2-default {
    color: #5f5d5d !important;
}
```
6. Transfer link text
  - Cannot reproduce in Vanilla Hyrax
  - `TODO`: Change text color settings in Nurax

I thought about a couple of options to fix 3,4,5:
- [Upgrading to bootstrap v4](https://medium.com/@sarahcodes/accessibility-issues-to-watch-out-for-in-bootstrap-4-962b993f0c64) will not fix these issues. 

- Add an additional hyrax_a11y.css file to the Hyrax application: That way even if we upgrade bootstrap at any point we will still continue to have these changes. Also, this can serve as a single point for all a11y changes, perhaps?

@mtribone @adamjarling  Thoughts?

**Comment by h-parekh** [comment]
![dashboard_page_contrast_on_hyrax_vanilla](https://user-images.githubusercontent.com/21689371/32850492-517499a6-ca00-11e7-8aef-ae4a254e9f33.png)

**Comment by mtribone** [comment]
@h-parekh As for the breadcrumbs foreground color , it appears that the color is being overridden in the dashboard style sheet to `#0080ff` which fails because it is 3.48:1. It looks like the order of the stylesheets is affecting the outcome because that link color for home is a lighter blue than what is in the WAVE output.

My screen shot from Nurax:

![screen shot 2017-11-15 at 1 07 58 pm](https://user-images.githubusercontent.com/4163828/32852591-656174c4-ca06-11e7-865d-1be762e3ec40.png)

**Comment by mtribone** [comment]
@h-parekh Aaahhh the link colors for the breadcrumbs and transfer are coming from inline styles. 


![screen shot 2017-11-15 at 1 15 00 pm](https://user-images.githubusercontent.com/4163828/32852782-062b64d2-ca07-11e7-96c8-2c54a3686eb1.png)

**Comment by mtribone** [comment]
I like the idea of a stylesheet specifically for a11y issues!

**Comment by h-parekh** [comment]
So the color for #2 (breadcrumb link color (Home <a> tag)) and #6 (Transfer link text) are both coming from inline styles. Both Vanilla Hyrax and Nurax are showing the same behavior. 

But on Vanilla Hyrax the breadcrumb link fails contrast ratio and the transfer link doesn't.

**Comment by h-parekh** [comment]
Link to the vanilla Hyrax instance i am looking at: https://a11y.demo.hydrainabox.org

**Comment by mtribone** [comment]
@h-parekh Hmmmm...looks like we have differing versions of the codebase. Nurax versus Hyrax versus Hyku.

**Comment by jcoyne** [comment]
Are the defaults a problem or is it how the user has chosen to customize the colors that is the problem?

<img width="1312" alt="screen shot 2017-11-15 at 1 15 45 pm" src="https://user-images.githubusercontent.com/92044/32855411-2b03d9a6-ca07-11e7-8188-06c5ecb4b579.png">

**Comment by h-parekh** [comment]
@mtribone: Aah, that's a rookie mistake. Hyku should map to a specific Hyrax release I think? 

@mjgiarlo Can you confirm the Hyrax version for https://a11y.demo.hydrainabox.org/?locale=en The footer says 'Hyku v1.0.0.beta1' but I don't know which version of Hyrax its running on. 

@jcoyne As far as I can tell from the analysis above, items 1,2 and 6 are due to customizations in Nurax and 3-5 are due to defaults in Hyku v1.0.0.beta1

**Comment by mjgiarlo** [comment]
Hyku is running a new enough Hyrax, @h-parekh.

**Comment by h-parekh** [comment]
@mtribone Help me understand this, if we just look at the Hyku dashboard, 'Home' link in breadcrumb  and 'Select works to transfer' link are using the same inline styling. But only the 'Home' link reports a contrast error. Am i missing something? 

<img width="1215" alt="screen shot 2017-11-15 at 2 41 16 pm" src="https://user-images.githubusercontent.com/21689371/32856622-4bc936de-ca13-11e7-825e-583a3b030ae1.png">


<img width="1226" alt="screen shot 2017-11-15 at 2 40 58 pm" src="https://user-images.githubusercontent.com/21689371/32856620-4bbc7296-ca13-11e7-918f-b5ee5550281c.png">

**Comment by jcoyne** [comment]
@h-parekh earlier you said:
> Upgrading to bootstrap v4 will not fix these issues.

but the article you posted references an early alpha release of bootstrap 4.  Have these issues been addressed since then?

**Comment by jcoyne** [comment]
@h-parekh 
> But only the 'Home' link reports a contrast error. Am i missing something?

the home link is on a darker (panel header) background.

**Comment by h-parekh** [comment]
@jcoyne 
Great catch! There is a very slight difference in the background colors. The Home link has a 7.5:1 contrast that's meeting AA and AAA standards so not sure why the WAVE tool still throws an error. 

As for the bootstrap v4 accessibility fixes, I just checked the documentation for the [current release v4.0](https://getbootstrap.com/docs/4.0/getting-started/accessibility/#color-contrast) and it says that:
`Authors will need to manually modify/extend these default colors to ensure adequate color contrast ratios`

**Comment by mtribone** [comment]
@jcoyne @h-parekh The tricky thing is that the foreground/background contrast ratios are also affected by text size. So for example, a color combo at 14px for text may not pass but at 20px it might.

**Comment by h-parekh** [comment]
@mtribone Really tricky stuff! Took much longer to get a PR in that I thought.

https://github.com/samvera/hyrax/pull/2205

**Comment by h-parekh** [comment]
Confirmed with @mtribone, this contrast issue has been fixed in master.
